### ISSUE TITLE
Remove descriptions from PropertyType and use Config.Description

### ISSUE THREAD

**Issue by afsalthaj** [zio/zio-config#19]
Depends on
  #18 

```scala
 trait Config[A] {
    def <?>(description: String): Config[A] =
      Config.Describe(self, description) 
}
```

And the new functions that made use of `PropertyType` descriptions will be

```scala
  def int(path: String): Config[Int] =
    Config.Source(path, PropertyType.IntType) <?> "value of type int"
```

This allows more generic descriptions to be tagged against the overall config (and its nodes)

**Comment by afsalthaj** [comment]
this is resolved in #64 

We have a new functionality `Config.int("PORT").default(80` similar to `Config.int("PORT").optional`

**Comment by afsalthaj** [comment]
This is resolved
